dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/state/BitList.java

```diff
@@ -93,23 +93,23 @@ public BitList(List<E> originList, BitSet rootSet, List<E> tailList) {
     }
 
     // Provided by BitList only
-    public List<E> getOriginList() {
+    public synchronized List<E> getOriginList() {
```

Comment:
Better to use a read and write lock?

Reply:
In general, reading without locking may lead to consistency issues.
@He-Pin If the operation itself is very short, using a read-write lock won't increase throughput. In fact, due to the internal complexity of the lock, it might even slow down. I've tested this with examples before.

Reply:
Great, at least at this case, this is not needed:)